packages/app/cypress/e2e/specs_page.cy.ts

```diff
@@ -0,0 +1,346 @@
+            cy.findByPlaceholderText(defaultMessages.components.fileSearch.byExtensionInput)
+            .as('ExtensionInput')
+            .clear()
+            .type('foobar')
+
+            cy.findByTestId('file-match-indicator').should('contain', 'No Matches')
+
+            cy.findByTestId('no-results-clear').click()
+
+            cy.get('@ExtensionInput').should('have.value', '**/*.{jsx,tsx}')
+
+            cy.findByTestId('file-match-indicator').should('contain', '2 Matches')
+          })
+        })
+
+        it('shows input for file name filter', () => {
+          cy.get('@CreateFromComponentDialog').within(() => {
+            cy.findByLabelText('file-name-input').as('FileNameInput')
+            .should('have.value', '')
+
+            cy.findByTestId('file-match-indicator').should('contain', '2 Matches')
+
+            cy.get('@FileNameInput')
+            .type('foobar')
+
+            cy.findByTestId('file-match-indicator').should('contain', 'No Matches')
+
+            cy.findByTestId('no-results-clear').click()
+
+            cy.findByTestId('file-match-indicator').should('contain', '2 Matches')
+
+            cy.get('@FileNameInput')
+            .type('App.jsx')
+
+            cy.findByTestId('file-match-indicator').should('contain', '1 of 2 Matches')
+          })
+        })
+
+        it('shows success modal with spec is created from component', () => {
+          cy.get('@CreateFromComponentDialog').within(() => {
+            cy.findAllByTestId('file-list-row').eq(0).as('NewSpecFile')
+
+            // TODO: assert visibility of secondary text on hover/focus when
+            // item is made keyboard accessible
+            // cy.get('@NewSpecFile).focus()
+            // cy.findByText('src/stories/Button.stories.jsx').should('be.visible')
+
+            cy.get('@NewSpecFile').click()
+          })
+
+          cy.findByRole('dialog', { name: defaultMessages.createSpec.successPage.header }).as('SuccessDialog').within(() => {
+            cy.validateExternalLink({ name: 'Need help?', href: 'https://on.cypress.io' })
+            cy.findByRole('button', { name: 'Close' }).should('be.visible')
+            cy.findByRole('link', { name: 'Okay, run the spec' }).should('have.attr', 'href', '#/specs/runner?file=src/App.cy.jsx')
+            cy.findByRole('button', { name: 'Create another spec' }).click()
+          })
+
+          // 'Create a new spec' dialog presents with options when user indicates they want to create
+          // another spec.
+          cy.findByRole('dialog', { name: defaultMessages.createSpec.newSpecModalTitle }).within(() => {
+            cy.findAllByTestId('card').eq(0)
+            .should('have.attr', 'tabindex', '0')
+
+            // the storybook card remains disabled here
+            cy.findAllByTestId('card').eq(1)
+            .should('have.attr', 'tabindex', '-1')
+          })
+        })
+      })
+    })
+  })
+
+  describe('Code Generation', () => {
```

Comment:
Full credit to @ZachJW34 for writing the Code Generation tests, I moved them into into this file to co-locate them with the rest of the specs page tests.